compiler/rustc_hir/src/lang_items.rs

```diff
@@ -139,6 +146,8 @@ pub fn extract(attrs: &[ast::Attribute]) -> Option<(Symbol, Span)> {
 
 language_item_table! {
 //  Variant name,            Name,                     Getter method name,         Target                  Generic requirements;
+    SyncDrop,                sym::sync_drop,           sync_drop_trait,            Target::Trait,          GenericRequirement::None;
+    Leak,                    sym::leak,                leak_trait,                 Target::Trait,          GenericRequirement::None;
```

Comment:
I suggest 1) not focusing on names of the lang items and 2) reserve more of such lang items.

I.e. have `default_trait1`, `default_trait2`, `default_trait3` and `default_trait4` instead of `leak` and `sync_drop`.

We probably need to resolve the issues with trait solver first, though.